components/LoginModal.tsx

```diff
@@ -0,0 +1,227 @@
+              return (
+                <>
+                  <div className="w-full">
+                    <Select
+                      label="Wallet to connect"
+                      value={selectedWallet}
+                      options={options}
+                      onChange={setSelectedWallet}
+                      onSearch={() => {}}
+                      withSearch={false}
+                      renderOption={({ value, label }) => {
+                        return (
+                          <div className="flex flex-col items-start overflow-hidden">
+                            <span>{label}</span>
+                            <span className="text-zinc-400">{value}</span>
+                          </div>
+                        );
+                      }}
+                    />
+                  </div>
+                  <div className="mt-4 flex items-center justify-center space-x-4">
+                    <button
+                      type="button"
+                      className="rounded border-2 bg-transparent px-3 py-1 font-medium text-white hover:outline-none"
+                      onClick={async e => {
+                        e.preventDefault();
+
+                        if (!state.p2pClient!.hasReceivedPermissionRequest())
+                          return;
+
+                        await state.p2pClient!.refusePermission();
+                        setCurrentState(State.REFUSED);
+                      }}
+                    >
+                      Refuse
+                    </button>
+                    <button
+                      type="button"
+                      className={
+                        "rounded border-2 border-primary bg-primary px-3 py-1 font-medium text-white hover:border-red-500 hover:bg-red-500 hover:outline-none focus:border-red-500 focus:bg-red-500"
+                      }
+                      onClick={async e => {
+                        e.preventDefault();
+
+                        if (
+                          !parsedData ||
+                          !state.p2pClient!.hasReceivedPermissionRequest() ||
+                          !selectedWallet
+                        )
+                          return;
+
+                        setCurrentState(State.LOADING);
+                        await state.p2pClient!.approvePermission(
+                          selectedWallet.value
+                        );
+                        setCurrentState(State.AUTHORIZED);
+                        dispatch({
+                          type: "addDapp",
+                          payload: {
+                            data: parsedData,
+                            address: selectedWallet.value,
+                          },
+                        });
+                      }}
+                    >
+                      Authorize
+                    </button>
+                  </div>
+                </>
+              );
+            case State.NO_AVAILABLE_WALLET:
```

Comment:
If dapp make a connection request to new browser, UI should be in this state as well.